policyengine_us_data/utils/loss.py

```diff
@@ -630,44 +672,62 @@ def _add_agi_state_targets():
             axis=1,
         )
     )
-
+    
     target_names = soi_targets["target_name"].tolist()
     target_values = soi_targets["VALUE"].astype(float).tolist()
     return target_names, target_values
 
 
-def _add_agi_metric_columns(
-    loss_matrix: pd.DataFrame,
+def add_agi_metric_columns(
+    loss_matrix: pd.DataFrame, 
     sim,
 ):
     """
-    Add AGI metric columns to the loss_matrix.
+    Add AGI & National Income tax metric columns by state to the loss_matrix from the SOI.
     """
     soi_targets = pd.read_csv(STORAGE_FOLDER / "agi_state.csv")
-
+    
+    # AGI is at tax unit level
     agi = sim.calculate("adjusted_gross_income").values
+    
+    # Federal income tax is at household level - need to map to tax unit from household
+    federal_income_tax_household = sim.calculate("federal_income_tax").values
+    federal_income_tax = sim.map_result(
+        federal_income_tax_household, "household", "tax_unit", how="divide_by_num_of_tax_units"
```

Comment:
federal income tax is actually already at tax unit level, no need for this! `income_tax` is what you're looking for